moto/ec2/models/instances.py

```diff
@@ -756,7 +757,17 @@ def _original_run_instances(
     # instances are never deleted (they are kept in "terminated" state)
     # so there is no clean-up to do
     def run_instances(self, *args, **kwargs):
-        if not hasattr(self, 'id_to_instances'):
+        # Because, it is not easy to reach an instance type failure/unavailability,
+        # we raise an error when a specific tags is passed on.
+        tags = kwargs.get("tags", {})
+        instance_tags = tags.get("instance", {})
+        if instance_tags.get("moto-error", None) == "unavailable_instance_type":
+            raise EC2ClientError(
+                "InstanceLimitExceeded",
+                "InstanceLimitExceeded: Your quota allows for 0 more running instance(s).",
+            )
+
+        if not hasattr(self, "id_to_instances"):
```

Comment:
Any changes below seem to be automatic changes done by your IDE. Can you remove them? or hand-paste the change above into a fresh copy? I ask this because I'd like to keep the delta between upstream and us to a minimum. Sooner or later we will certainly need to rebase.

Reply:
Thanks!